TCAT/Controllers/HomeMapViewController.swift

```diff
@@ -0,0 +1,251 @@
+        indicator.snp.makeConstraints { (make) in
+            make.center.equalToSuperview()
+            make.size.equalTo(loadingIndicatorSize)
+        }
+        
+    }
+    
+    func removeLoadingScreen() {
+        loadingView.removeFromSuperview()
+        viewWillAppear(false)
+    }
+}
+
+// MARK: Location Delegate
+extension HomeMapViewController: CLLocationManagerDelegate {
+    
+    func locationManager(_ manager: CLLocationManager, didChangeAuthorization status: CLAuthorizationStatus) {
+        
+        if status == .denied {
+            let alertTitle = Constants.Alerts.LocationDisabled.title
+            let alertMessage = Constants.Alerts.LocationDisabled.message
+            let alertController = UIAlertController(title: alertTitle, message: alertMessage, preferredStyle: .alert)
+            let settingsAction = UIAlertAction(title: Constants.Alerts.LocationDisabled.settings, style: .default) { (_) in
+                UIApplication.shared.open(URL(string: UIApplication.openSettingsURLString)!, options: convertToUIApplicationOpenExternalURLOptionsKeyDictionary([:]), completionHandler: nil)
+            }
+            
+            guard let showReminder = userDefaults.value(forKey: Constants.UserDefaults.showLocationAuthReminder) as? Bool else {
+                userDefaults.set(true, forKey: Constants.UserDefaults.showLocationAuthReminder)
+                let cancelAction = UIAlertAction(title: Constants.Alerts.LocationDisabled.cancel, style: .default, handler: nil)
+                alertController.addAction(cancelAction)
+                alertController.addAction(settingsAction)
+                alertController.preferredAction = settingsAction
+                present(alertController, animated: true)
+                return
+            }
+            
+            if !showReminder {
+                return
+            }
+            
+            let dontRemindAgainAction = UIAlertAction(title: Constants.Alerts.LocationDisabled.cancel, style: .default) { _ in
+                self.userDefaults.set(false, forKey: Constants.UserDefaults.showLocationAuthReminder)
+            }
+            alertController.addAction(dontRemindAgainAction)
+            
+            alertController.addAction(settingsAction)
+            alertController.preferredAction = settingsAction
+            
+            present(alertController, animated: true)
+            
+        }
+    }
+    
+    func locationManager(_ manager: CLLocationManager, didUpdateLocations locations: [CLLocation]) {
+        guard let location = locations.last else { return }
+        currentLocation = location
+    }
+    
+}
+
+extension HomeMapViewController: GMSMapViewDelegate {
+    func mapView(_ mapView: GMSMapView, willMove gesture: Bool) {
+        optionsCardVC.searchBar.endEditing(true)
+    }
+}
+
+extension HomeMapViewController: HomeOptionsCardDelegate {
+    func getCurrentLocation() -> CLLocation? { return currentLocation }
```

Comment:
are one line functions standard? @kevinchan159

Reply:
It's necessary to pass our currentLocation to other VCs